sdk/js/src/User.ts

```diff
@@ -26,13 +26,13 @@ export class DVCPopulatedUser implements DVCUser {
     readonly sdkType: 'client' | 'server'
     readonly sdkVersion: string
 
-    constructor(user: DVCUser, options: DVCOptions, staticData?: StaticData) {
+    constructor(user: DVCUser, options: DVCOptions, staticData?: StaticData, anonymousUserId?: string) {
         if (!user.user_id && !user.isAnonymous) {
             throw new Error('Must have a user_id, or have "isAnonymous" set on the user')
         }
 
-        this.user_id = !user.user_id ? uuidv4() : user.user_id
-        this.isAnonymous = user.user_id ? false : user.isAnonymous
+        this.user_id =  user.isAnonymous ? anonymousUserId || uuidv4() : user.user_id || uuidv4()
+        this.isAnonymous = user.isAnonymous || false
```

Comment:
blah || false always evaluates to blah, the extra false is redundant
```suggestion
        this.isAnonymous = user.isAnonymous
```

Reply:
I added this caz a test case `should make a new user if user id provided but no isAnonymous flag` was failed, should I update the test case then?

Reply:
the test `expect(user.isAnonymous).toBe(false)`, but if I don't make a default value, `user.isAnonymous` will be undefined

Reply:
ohh, i think because our tests are js file, so it didn't do type check... and actually adding the false value is also trying to follow the android sdk behaviour